### ISSUE TITLE
Sponge vulnerable to MITM

### ISSUE THREAD

**Issue by arkadiyt** [lobsters/lobsters#434]
`Sponge` is used to fetch url content in several places:
- handling github/twitter oauth flows (particularly sensitive)
- sending push notifications via pusher
- fetching story attributes
- fetching user avatars
- handling story caching with diffbot

However it has the following code: https://github.com/lobsters/lobsters/blob/master/extras/sponge.rb#L154
```
host.verify_mode = OpenSSL::SSL::VERIFY_NONE
```
which makes it vulnerable to MITM attacks.

**Comment by jcs** [comment]
> edit: I believe you are doing this because you are directly hitting an ip address to avoid server side request forgery issues

Partially, but also because it's fetching content from arbitrary websites where SSL validation doesn't matter and trying to enforce validation would break title fetching more often than a MITM would accomplish anything.

It should probably enable validation by default but allow disabling it from a setting, which would only get set during title fetching.

**Comment by evexoio** [comment]
@pushcx @alanpost can we close this issue? there doesn't look like any action needs to be taken here.

**Comment by evexoio** [comment]
> We rarely if ever see this on Lobster.rs, probably since no one wants to follow such links.

So is it really an issue? what is HN's take on this one btw?

**Comment by arkadiyt** [comment]
> So is it really an issue?

Yes, because the same http library is being used for github and twitter oauth flows - anyone on the network path between Lobste.rs and Twitter/Github can intercept those credentials and gain the same level of account access.

> what is HN's take on this one btw?

HN doesn't fetch anything from submitted urls.

**Comment by evexoio** [comment]
I see, then we should add https support for oauth flows and leave the title fetching without https.

let's see what @pushcx or @alanpost think about this.

**Comment by jbbarth** [comment]
This issue is a bit old but I think it deserves fixing. Everybody could agree I think that hitting trusted APIs should be done in HTTPS, even more if you transmit credentials/secrets like done in the GitHub OAuth flow. I gave a shot at a solution with good defaults in #692.

**Comment by jcs** [comment]
#692 was merged so this can be closed.

**Comment by pushcx** [comment]
Thanks for catching the stale issue, closing.
